modules/pipeline/pipengine/actionexecutor/manager.go

```diff
@@ -73,11 +107,99 @@ func (m *Manager) Get(name types.Name) (types.ActionExecutor, error) {
 	if len(name) == 0 {
 		return nil, errors.Errorf("executor name is empty")
 	}
+	m.RLock()
 	e, ok := m.executors[name]
+	m.RUnlock()
+	if ok {
+		return e, nil
+	}
+	m.RLock()
+	kind, ok := m.kinds[name]
+	m.RUnlock()
 	if !ok {
-		return nil, errors.Errorf("not found action executor [%s]", name)
+		return nil, errors.Errorf("executor name [%s] not found", name)
 	}
-	return e, nil
+	m.RLock()
+	create, ok := m.factory[kind]
+	m.RUnlock()
+	if !ok {
+		return nil, errors.Errorf("executor kind [%s] not found", kind)
+	}
+	actionExecutor, err := create(name, nil)
+	if err != nil {
+		return nil, errors.Errorf("executor [%s] created failed, err: %v", name, err)
+	}
+	m.Lock()
+	m.executors[actionExecutor.Name()] = actionExecutor
+	m.Unlock()
+	return actionExecutor, nil
+}
+
+func (m *Manager) GetCreateByKind(kind types.Kind) (types.CreateFn, error) {
+	if len(kind) == 0 {
+		return nil, errors.Errorf("executor kind is empty")
+	}
+	m.Lock()
+	f, ok := m.factory[kind]
+	m.Unlock()
+	if !ok {
+		return nil, errors.Errorf("unregistered action executor kind: [%s]", kind)
+	}
+	return f, nil
+}
+
+func (m *Manager) deleteK8SExecutor(cluster apistructs.ClusterInfo) {
```

Comment:
split to a new file for k8s related functions.

Reply:
adjusted